### PR TITLE
Save the pictures to the database

### LINKED ISSUE #59: Link photo taking ability in EntryActivity to database

### PR DESCRIPTION
**Pull request by catfinch**
Closes #59

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -60,4 +60,7 @@
     <string name="_speaker">*speaker*</string>
     <string name="_date">*date*</string>
 
+    <!-- Intent item key strings -->
+    <string name="intent_word_id_key">word id</string>
```
Comment (alpha0010) [inline app/src/main/res/values/strings.xml]: What is this for?

File: app/src/main/java/org/sil/gatherwords/EntryFragment.java [hunk 2]
```diff
@@ -122,4 +143,40 @@ public View getView(int position, View convertView, @NonNull ViewGroup parent) {
...
+            }
+
+            return wd.get(wordIDs[0]);
+        }
+
+        @Override
+        protected void onPostExecute(Word word) {
+            super.onPostExecute(word);
+            View entryPage = entryPageRef.get();
+            ImageView picture = entryPage.findViewById(R.id.pic_viewer);
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/EntryFragment.java]: Potential npe from the weak ref.

File: app/src/main/java/org/sil/gatherwords/EntryFragment.java [hunk 3]
```diff
@@ -122,4 +143,40 @@ public View getView(int position, View convertView, @NonNull ViewGroup parent) {
...
+
+        UpdateAfterPicture(View entryPage) {
+            entryPageRef = new WeakReference<>(entryPage);
+            wd = AppDatabase.get(entryPage.getContext()).wordDao();
+        }
+
+        @Override
+        protected Word doInBackground(Long... wordIDs) {
+            if (wordIDs == null || wordIDs.length != 1)  {
+                Log.e(LoadWordTask.class.getSimpleName(), "Did not receive exactly 1 wordID");
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/EntryFragment.java]: This is `UpdateAfterPicture` not `LoadWordTask`.

File: app/src/main/java/org/sil/gatherwords/EntryActivity.java [hunk 4]
```diff
@@ -218,9 +223,39 @@ protected void onPostExecute(List<Long> wordIDs) {
...
-    // Camera/Image processing
-    // based on https://developer.android.com/training/camera/photobasics.html
+        private final AppDatabase db;
+        private final WordDao wordDao;
+        private final long wordId;
+
+        StorePicturesTask(EntryActivity activity) {
+            db = AppDatabase.get(activity.getApplicationContext());
+            wordDao = db.wordDao();
+            EntryFragment fragment = (EntryFragment)((EntryPagerAdapter) activity.pager.getAdapter()).getItem(activity.pager.getCurrentItem());
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/EntryActivity.java]: Seems like a candidate to pull into a method.  More usable if we could call something like `activity.getCurrrentEntryFragment()`, since it sounds like we will be reusing this code.

File: app/src/main/java/org/sil/gatherwords/EntryActivity.java [hunk 5]
```diff
@@ -218,9 +223,39 @@ protected void onPostExecute(List<Long> wordIDs) {
         }
     }
 
+    private static class StorePicturesTask extends AsyncTask<Bitmap, Void, Void> {
 
-    // Camera/Image processing
-    // based on https://developer.android.com/training/camera/photobasics.html
+        private final AppDatabase db;
+        private final WordDao wordDao;
+        private final long wordId;
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/EntryActivity.java]: Why are these final?

File: app/src/main/java/org/sil/gatherwords/EntryFragment.java [hunk 6]
```diff
@@ -61,6 +63,25 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+    }
+
+    /**
+     * Gets the current wordId
+     *
+     * Sometimes m_wordID is not yet set for some reason
+     * @return The wordID of the current Word
+     */
+    public long getWordID() {
+        return getArguments().getLong(ARG_WORD_ID);
```
Comment (alpha0010) [inline app/src/main/java/org/sil/gatherwords/EntryFragment.java]: Why is this necessary?  In what situation is `m_wordID` not set?

File: app/src/main/res/layout/fragment_entry.xml [hunk 7]
```diff
@@ -16,6 +16,16 @@
...
+
+    </ListView>
+
+    <ImageView
+        android:id="@+id/pic_viewer"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:adjustViewBounds="true"
+        android:scaleType="centerInside"
+        android:src="@android:drawable/ic_menu_camera" />
```
Comment (alpha0010) [inline app/src/main/res/layout/fragment_entry.xml]: Recommend loading this as the final element in the `ListView` to make it scrollable.  Otherwise we may end up with very little screen space for text fields.  (In the future, it would probably be useful to also place the audio playback as a  `ListView` entry.)

File: app/src/main/java/org/sil/gatherwords/EntryFragment.java (hunk 6 above)
Comment (catfinch) [reply app/src/main/java/org/sil/gatherwords/EntryFragment.java]: When I first made this, after figured out how to get the correct fragment, I kept getting a `0` when I should not have. I'll try again though

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (catfinch) [reply app/src/main/res/values/strings.xml]: Left over from a different attempt to get the word id
Will remove

File: app/src/main/res/layout/fragment_entry.xml (hunk 7 above)
Comment (catfinch) [reply app/src/main/res/layout/fragment_entry.xml]: I tried that for the same reason but it would crash

File: app/src/main/java/org/sil/gatherwords/EntryFragment.java (hunk 6 above)
Comment (alpha0010) [reply app/src/main/java/org/sil/gatherwords/EntryFragment.java]: Not certain why this occurs, but it seems to be only upon returning from taking a photo.  Probably update the comment to specify this, and call it good enough.

File: app/src/main/res/layout/fragment_entry.xml (hunk 7 above)
Comment (alpha0010) [reply app/src/main/res/layout/fragment_entry.xml]: TODO for another pr.

**Comment by alpha0010** [comment]
Merged in 2f5c35f98d4c94abe7991c12c78a62c67cc545c4.
